tests/test_graph.py

```diff
@@ -1,98 +1,115 @@
 from __future__ import annotations as _annotations
 
+from dataclasses import dataclass
 from datetime import timezone
 from typing import Union
 
 import pytest
 from inline_snapshot import snapshot
 
-from pydantic_ai_graph import BaseNode, End, Graph, GraphContext, Snapshot
+from pydantic_ai_graph import BaseNode, End, EndEvent, Graph, GraphContext, Step
 
 from .conftest import IsFloat, IsNow
 
 pytestmark = pytest.mark.anyio
 
 
 async def test_graph():
-    class Float2String(BaseNode[None, float]):
+    @dataclass
+    class Float2String(BaseNode[None]):
```

Comment:
```suggestion
    class Float2String(BaseNode):
```
Elegantly, we can remove the state generic param in these cases.